### ISSUE TITLE
Feature request: the last player without a fold has the choice to show cards or not

### ISSUE THREAD

**Issue by NitroLine** [kuco23/pokerlib#3]
Hi, want implenet this feature : the last player without a fold has the choice to show cards or not. To use this need add PLAYERACTIONREQUIRED when DECLAREPREMATUREWINNER action appeare

**Comment by kuco23** [comment]
Hi, I think the best way to do this would be to overwrite [this](https://github.com/kuco23/pokerlib/blob/master/pokerlib/_round.py#L191) function into one that just calls PLAYERACTIONREQUIRED. Then another one should be added - one with a similar implementation as `_dealPrematureWinnings` but gets called when a user replies to PLAYERACTIONREQUIRED. [This](https://github.com/kuco23/pokerlib/blob/master/pokerlib/_round.py#L373) function should be modified to accept the new (input id)[https://github.com/kuco23/pokerlib/blob/master/pokerlib/enums.py#L53] and somehow try to verify if it is really being called from a user that was a premature winner at this round instance.

I think this is a very "internal" way of doing it, but doing this at the table layer seems to get ugly.

**Comment by NitroLine** [comment]
#4

**Comment by kuco23** [comment]
Hi, sorry about this, but when I was merging your request I realized that rewriting round for this feature would make Round a bit non-general, as with some poker implementations cards can be hidden by losing players also. So, I thought it would be better to write this feature into a separate Round implementation. I've made [this](https://github.com/kuco23/pokerlib/blob/3-decide-whether-to-show-cards-when-unnecessary/pokerlib/table_implementations/table_with_choice_to_show_cards.py) implementation for this feature. Note that it's not yet tested.

Also IO enums had to be made more dynamic to allow for extensions.
